core/src/main/java/hudson/XmlFile.java

```diff
@@ -205,9 +206,20 @@ public String toString() {
      * Opens a {@link Reader} that loads XML.
      * This method uses {@link #sniffEncoding() the right encoding},
      * not just the system default encoding.
+     * @throws IOException Encoding issues
+     * @return Reader for the file. should be close externally once read.
      */
     public Reader readRaw() throws IOException {
-        return new InputStreamReader(new FileInputStream(file),sniffEncoding());
```

Comment:
+1 and I'm not sure we should catch IO Exception inside, as it's already expected from the caller to close the stream.